acsylla/factories.py

```diff
@@ -38,10 +40,17 @@ def create_cluster(
     If `consistency` is provided the default value would be override, any statment will use
     by default that consistency level unless it is specificily configured at statement level.
     """
+    if username is None:
+        username = ""
+    if password is None:
+        password = ""
```

Comment:
Im not convinced with that, any chance to send the objects to the `_cython.cyacsylla.Cluster` constructor and then once there do the following check?

```python
if username is not None and password is not None:
   cass_cluster_set_credentials(.......)
elif username is not None or password is not None:
    raise ValueError("For using credentials both parametres need to be set")
```
By doing this we make IMO the credentials setup more clear, and anything different a None in both parameters is an attempt of the user for using a credentials, anything else is considered either a ValueError (only when one of them are set) or just a none configuration of the credentials.

This would also require a unit test then

Reply:
@pfreixes yeah, fixed